### PR TITLE
Add kaiko client

### PR DESCRIPTION
**Pull request by AymericNoel**
Implementation of 3 endpoints : 
+  one to get all supported exchanges
+  one to get all supported currencies
+  one to get last price for a given pair

### CODE REVIEW

File: src/Kaiko-Api-Client/openApi3.yaml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+info:
+  title: Kaiko API
+  version: '1.0.0'
+servers:
+  - url: 'https://eu.market-api.kaiko.io/'
+    description: Kaiko base url
+  - url: 'https://reference-data-api.kaiko.io/v1/'
+    description: Override base path for all operations with the /Instruments path
+paths:
+  https://reference-data-api.kaiko.io/v1/instruments:
```
Comment (monsieurleberre) [inline src/Kaiko-Api-Client/openApi3.yaml:11]: can you make that path relative?

**Comment by AymericNoel** [comment]
I've seen your publish nuget file. I need to rename the projects

**Review by monsieurleberre — CHANGES_REQUESTED** [review]
Looks good, the namespace project etc, should not really include underscores, on other projects we try to do market-making => MarketMaking, or trakx-tools => TrakxTools.
Can you please change that to conform ?  Also that will give you nuget packaging for free ;) you can look at the workflow flie to see why 👍 
I also think there should be several clients, even within ReferenceDataClient and MarketDataClient. 
image
here maybe on per sub level (might have to resolve a conflict between the 2 general clients but maybe it won't appear yet)

File: src/Kaiko-Api-Client/openApi3.yaml (hunk 1 above)
Comment (AymericNoel) [reply src/Kaiko-Api-Client/openApi3.yaml:11]: At the begginning, it was relative, but I didn't find a way to select multiple base path. 
Instruments and exchanges have the same base path but it's not the same as the one for getVwap.

And Instruments and exchanges are useful to get the supported exchanges and currencies to put in parameter of getVwap...

Comment (monsieurleberre) [reply src/Kaiko-Api-Client/openApi3.yaml:11]: OK let me have a look :) we might need to have 2 configurators and select the one with correct base path on injection, or have 2 swagger files...

Comment (monsieurleberre) [reply src/Kaiko-Api-Client/openApi3.yaml:11]: I don't know what is best, to have simply 2 swagger files or to go through the pain of adding url on each path...
probably 2 swagger files is OK

Comment (AymericNoel) [reply src/Kaiko-Api-Client/openApi3.yaml:11]: Yes, I've seen these links yesterday but impossible to generates different bath path. I look into all issues in nswag by Rico but didn't find anything else...

I think that I'm gonna make two swagger files...

**Review by monsieurleberre — APPROVED** [review]
Changed:
- project paths
- swagger filenames and MarketData tag
- add common csproj and packable .csproj to get nugets
- use TT to generate client classes
- Experimental: use serilog rather than microsoft logging, users of the client have to do the configuration in their library

**Comment by AymericNoel** [comment]
Great !

**Pull request merged**